proto/store/types.proto

```diff
@@ -0,0 +1,9 @@
+syntax = "proto3";
+package tendermint.proto.store;
+
+option go_package = "github.com/tendermint/tendermint/proto/store";
+
+message BlockStoreStateJSON {
```

Comment:
Maybe we should just store this as Protobuf rather than JSON? It's stored in a binary database anyway. It would be a breaking change (which I guess all of this is anyway), but if necessary we could support reading the old JSON and converting to Protobuf for the next save.